### DISCUSSION TITLE
getStaticProps slug param returning favicon.ico

### DISCUSSION THREAD

**Discussion by VictorPietro** [vercel/next.js#29082] [Help] [answered]
Hello, I have this application where I'm building something similar to Linktree.

When I enter in a /user/page/[slug], I want to get the slug of the page I have just entered, so I send this slug to my API (using AXIOS), and fetch every link bond to this page.

To get the slug, I am using getStaticProps, since I need this data before it's loaded.

But there is an unexpected/strange behavior going on (I am new to next.js, sorry if it is a dumb problem).

Let's say I enter in /user/page/example. When I console.log(params.slug), it prints "example", which is correct, then sends a GET request and fetches all of the "example" page links.

Until this part it is working as expected, but then, shortly thereafter, for some reason, console.log() prints again - and it prints "favicon.ico", as if it was my page slug. And then, it tries to fetch a page with "favicon.ico" url, and obviously doesn't find any page with this slug. This behavior repeats one last time.

In summary: I get page slug, send to API, it returns what I want, but then shortly thereafter my slug is turned into "favicon.ico" for unknown reasons, it sends it to API once again - as if it was my page slug, which it isn't -, and now, evidently, my request is wrong. And it repeats one more time. So, 1. correctly get slug (should stop here), 2. incorrectly get "favicon.ico" as slug, 3. incorrectly get "favicon.ico" (not sure why it happens again) and then it stops.

Only place "favicon.ico" is located is in the <Head> tag:

```
// _document.tsx

import Document, { Html, Head, Main, NextScript } from 'next/document';

export default class MyDocument extends Document {
    render() {
        return (
            <Html>
                <Head>
                    <link rel="preconnect" href="https://fonts.gstatic.com" />
                    <link href="https://fonts.googleapis.com/css2?family=Roboto:wght@400;700;900&display=swap" rel="stylesheet" />
                    <link rel="shortcut icon" href="favicon.ico" />
                </Head>
                <body>
                    <Main />
                    <NextScript />
                </body>
            </Html>
        );
    }
}
```

And here is my getStaticProps call:

```
// /pages/user/page/[slug].tsx

export const getStaticProps: GetStaticProps = async ({ params }) => {
    let slug = '';
    let links = {};
    let page = {};

    try {
        slug = Array.isArray(params.slug) ? params.slug[0] : params.slug

        const response = await api.get('links', {
            params: {
                page_url: slug
            }
        });

        const links = response.data.map(link => {
            return {
                title: link.title,
                description: link.description,
                url: link.url,
                updatedAt: new Date(link.updated_at).toLocaleDateString('pt-BR', {
                    day: '2-digit',
                    month: 'long',
                    year: 'numeric'
                })
            }
        });
    } catch (err) {
        console.log(err.response)
    }

    return {
        props: {
            slug,
            links,
            page
        },
        revalidate: 60 * 30,    // 30 minutes
    }
}
```

Thanks in advance.

**Comment by wwwdepot** [comment, marked as answer]
Use absolute URL for favicon! In your case, it loads favicon at the page location  `/user/page/favicon.ico` instead of your public folder.
```js
// fix
<link rel="shortcut icon" href="/favicon.ico" />
```

**Reply by VictorPietro** [reply]
It was way foolisher than I thought lol. I have been on this for days. Thank you very much @wwwdepot.
